lib/sql-logging/configuration.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+module SqlLogging
+  class Configuration
+    DEFAULTS = {
+      show_sql_backtrace: true,
+      show_top_sql_queries: :total_time,
+      top_sql_queries: 10,
+      backtrace_cleaner: nil,
+      logger: nil
+    }.freeze
+
+    attr_accessor(*DEFAULTS.keys)
+
+    def initialize
+      DEFAULTS.each do |k, v|
+        instance_variable_set("@#{k}", v)
+      end
+    end
+
+    def show_top_sql_queries=(value)
+      validate_allowed([false, :rows, :queries, :bytes, :total_time,
+                        :median_time], value, __method__)
+    end
+
+    def backtrace_cleaner
+      @backtrace_cleaner ||= initialize_cleaner
+    end
+
+    def logger
+      @logger ||= Rails.logger
+    end
+
+    private
+
+    def initialize_cleaner
+      Rails.backtrace_cleaner.dup.tap do |cleaner|
+        cleaner.add_silencer { |line| line =~ %r{sql-logging/lib} }
```

Comment:
Should setting the `backtrace_cleaner` lead to this line to be appended in some way so the user does not have to keep this on their end?